leapp/actors/__init__.py

```diff
@@ -13,73 +13,130 @@
+    define :ref:`messages <terminology:message>` the actor consumes.
+    """
+
     produces = ()
+    """
+    Tuple of :py:class:`leapp.models.Model` derived classes defined in the :ref:`repositories <terminology:repository>` that define
+    :ref:`messages <terminology:message>` the actor produces.
+    """
+
     tags = ()
+    """
+    Tuple of :py:class:`leapp.tags.Tag` derived classes by which :ref:`workflow <terminology:workflow>` :ref:`phases <terminology:phase>` select
+    actors for execution.
+    """
 
     def __init__(self, messaging=None, logger=None):
         self._messaging = messaging
         self.log = (logger or logging.getLogger('leapp.actors')).getChild(self.name)
-        """ Configured logger for the actor """
+        """ A configured logger instance for the current actor """
 
     @property
     def actor_files_paths(self):
+        """ Returns the file paths that are bundled with the actor. (Path to the content of the actor's file directory)"""
         return os.getenv("LEAPP_FILES", "").split(":")
 
     @property
     def files_paths(self):
+        """ Returns all actor file paths (related ones to the actor and common actors files paths). """
         return self.actor_files_paths + self.common_files_paths
 
     @property
     def common_files_paths(self):
+        """ Returns all common repository file paths. """
         return os.getenv("LEAPP_COMMON_FILES", "").split(":")
 
     def get_folder_path(self, name):
+        """
+        Displays first folder path within actor.
+
+        :param name: Name of the folder
+        :type name: str
+        :return: Found folder path
+        :rtype: str or None
+        """
         for path in self.files_paths:
             path = os.path.join(path, name)
             if os.path.isdir(path):
                 return path
         return None
 
     def get_file_path(self, name):
+        """
+        Displays first file path within actor and common actors file paths.
+
+        :param name: Name of the file
+        :type name: str
+        :return: Found file path
+        :rtype: str or None
+        """
         for path in self.files_paths:
             path = os.path.join(path, name)
             if os.path.isfile(path):
                 return path
         return None
 
     def run(self, *args):
+        """ Runs actor calling method :py:func:`process`. """
         os.environ['LEAPP_CURRENT_ACTOR'] = self.name
         try:
             self.process(*args)
         finally:
             os.environ.pop('LEAPP_CURRENT_ACTOR', None)
 
     def process(self, *args, **kwargs):
+        """ Main processing method (in inherited actors, the function needs to be defined to be able process)"""
         raise NotImplementedError()
 
-    def produce(self, *args):
+    def produce(self, *models):
+        """
+        After running :py:func:`process`, it prepares processed data - results to next processing in workflow.
```

Comment:
No this is not correct it is immediately when produce is called.
User rather this:

By calling produce model instances are store as messages. Those messages can be then consumed by other actors.